platform/podjs/src/browserPod.ts

```diff
@@ -15,68 +15,130 @@ import * as zip from "@zip.js/zip.js";
 import endpointsJson from "../../../../polyPod-config/endpoints.json";
 import { Manifest, readManifest } from "./manifest";
 
+const DB_NAME = "polypod";
+const DB_VERSION = 1;
+const OBJECT_STORE_POLY_IN = "poly-in";
+const OBJECT_STORE_POLY_OUT = "poly-out";
+
 const NAV_FRAME_ID = "polyNavFrame";
 const NAV_DEFAULT_BACKGROUND_COLOR = "#ffffff";
 const NAV_DARK_FOREGROUND_COLOR = "#000000";
 const NAV_LIGHT_FOREGROUND_COLOR = "#ffffff";
 
-class LocalStoragePolyIn implements PolyIn {
-    private static readonly storageKey = "polyInStore";
-    private store = JSON.parse(
-        localStorage.getItem(LocalStoragePolyIn.storageKey) || "[]"
-    );
+async function migrateData(db: IDBDatabase): Promise<IDBDatabase> {
+    const polyInData = localStorage.getItem("polyInStore");
+    const polyOutData = localStorage.getItem("files");
+
+    if (polyInData || polyOutData) {
+        const quads = polyInData ? JSON.parse(polyInData) : [];
+        const files: FileInfo[] = [];
+
+        if (polyOutData) {
+            for (const [, file] of JSON.parse(polyOutData)) {
+                const dataUrl = localStorage.getItem(file.id);
```

Comment:
is there any chance that this will throw an error? How are we handling it?

Reply:
Not that I'm aware of. Am I missing anything?

Reply:
You're parsing a string. `polyOutData` can be corrupted. It can be other than an array, thus not parseable into an array of arrays, one of which has an `id` as an attribute.

Reply:
I don't think we have to be robust against corrupted _localStorage_ here - this code pretty much just runs for our devs - and only once per dev/feature, I figure.

Reply:
Gotya, this comment were meant to be on the line above where we parse the data to be migrated from `loacalStorage`. If that fails, the promise returned by `migrateData` will be rejected with the error thrown by `JSON.parse` and propagates to `openDatabase` and eventually to the methods of `PolyIn` and `PolyOut`. Also the code that clears `localStorage` won't be reached then. So yeah, I'd say this error scenario is handled as solid as it gets.